File: springboot_api/src/main/java/com/example/membership_api/model/AddressBook.java
package com.example.membership_api.model;


import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Table;
import lombok.Data;
import lombok.EqualsAndHashCode;

@Entity
@Data
@EqualsAndHashCode(callSuper = true)
@Table(name = "address_book")
public class AddressBook extends BaseModel {
    private String name;

    @Column(name = "user_id", nullable = false)
    private String userId;

    private String pincode;

    @Column(columnDefinition="TEXT")
    private String address;

    private String city;

    private String state;

    private String country;

    @Column(name = "is_default")
    private Boolean isDefault = true;
}

File: flask_api/app/models/address_book.py
from app.models.base import BaseModel, db


class AddressBook(BaseModel):
    __tablename__ = "address_book"

    name = db.Column(db.Text, nullable=False)
    user_id = db.Column(db.String(36), nullable=False)
    pincode = db.Column(db.String(36), nullable=False)
    address = db.Column(db.Text, nullable=False)
    city = db.Column(db.String(36), nullable=False)
    state = db.Column(db.String(36), nullable=False)
    country = db.Column(db.String(36), nullable=False)
    is_default = db.Column(db.Boolean, default=True)

    def __repr__(self):
        return f"<AddressBook {self.address}>"
